src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java

```diff
@@ -31,35 +43,183 @@ public static void addNodeToCanvas(NodeInfo node) {
+      Helper.locateElementByCssSelector(
+        Helper.getCssSelectorByDataTestId("pipeline-zoom-in-control"))
+    );
+  }
+
+  public static void closeConfigPopover() {
+    ElementHelper.clickOnElement(
+      Helper.locateElementByCssSelector(
+        Helper.getCssSelectorByDataTestId("close-config-popover"))
+    );
+  }
+
+  public static List<WebElement> findElementsByCssSelector(String cssSelector) {
+    return SeleniumDriver.getDriver()
+      .findElements(By.cssSelector(cssSelector));
+  }
+
+  public static void createSimplePipeline() {
+    NodeInfo sourceNode = new NodeInfo("Spanner", "batchsource", "0");
+    NodeInfo transformNode = new NodeInfo("Wrangler", "transform", "1");
+    NodeInfo sinkNode = new NodeInfo("BigQueryMultiTable", "batchsink", "2");
+
+    addNodeToCanvas(sourceNode);
+
+    toggleTransformPanel();
+    addNodeToCanvas(transformNode);
+    toggleTransformPanel();
+
+    toggleSinkPanel();
+    addNodeToCanvas(sinkNode);
+    toggleSinkPanel();
+
+    connectTwoNodes(sourceNode, transformNode);
+    connectTwoNodes(transformNode, sinkNode);
+  }
+
+  public static void createComplexPipeline() {
+    // Two BigQuery sources
+    NodeInfo sourceNode1 = new NodeInfo("Spanner", "batchsource", "0");
+    NodeInfo sourceNode2 = new NodeInfo("Spanner", "batchsource", "1");
+
+    // Two javascript transforms
+    NodeInfo transformNode1 = new NodeInfo("JavaScript", "transform", "2");
+    NodeInfo transformNode2 = new NodeInfo("JavaScript", "transform", "3");
+
+    // One joiner
+    NodeInfo joinerNode = new NodeInfo("Joiner", "batchjoiner", "4");
+
+    // One condition node
+    NodeInfo conditionNode = new NodeInfo("Conditional", "condition", "5");
+
+    // Two BigQuery sinks
+    NodeInfo sinkNode1 = new NodeInfo("BigQueryMultiTable", "batchsink", "6");
+    NodeInfo sinkNode2 = new NodeInfo("BigQueryMultiTable", "batchsink", "7");
+
+    addNodeToCanvas(sourceNode1);
+    addNodeToCanvas(sourceNode2);
+
+    toggleTransformPanel();
+    addNodeToCanvas(transformNode1);
+    addNodeToCanvas(transformNode2);
+    toggleTransformPanel();
+
+    toggleAnalyticsPanel();
+    addNodeToCanvas(joinerNode);
+    toggleAnalyticsPanel();
+
+    toggleConditionsAndActionsPanel();
+    addNodeToCanvas(conditionNode);
+    toggleConditionsAndActionsPanel();
+
+    toggleSinkPanel();
+    addNodeToCanvas(sinkNode1);
+    addNodeToCanvas(sinkNode2);
+    toggleSinkPanel();
+
+    pipelineCleanUpGraphControl();
+    fitPipelineToScreen();
+
+    connectTwoNodes(sourceNode1, transformNode1);
+    connectTwoNodes(sourceNode2, transformNode2);
+
+    connectTwoNodes(transformNode1, joinerNode);
+    connectTwoNodes(transformNode2, joinerNode);
+
+    connectTwoNodes(joinerNode, conditionNode);
+
+    connectTwoNodes(conditionNode, sinkNode1);
+    connectTwoNodes(conditionNode, sinkNode2);
+
+    pipelineCleanUpGraphControl();
+    fitPipelineToScreen();
+  }
+
+  public static void takeAScreenshot() {
```

Comment:
This function is more of an assistant purpose, which helps to have a viewable screenshot at a testing step